lib/ansible/modules/network/exos/exos_facts.py

```diff
@@ -0,0 +1,452 @@
+#!/usr/bin/python
+#
+# (c) 2018 Extreme Networks Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: exos_facts
+version_added: "2.7"
+author: "Lance Richardson (@hlrichardson)"
+short_description: Collect facts from devices running Extreme EXOS
+description:
+  - Collects a base set of device facts from a remote device that
+    is running EXOS.  This module prepends all of the base network
+    fact keys with C(ansible_net_<fact>).  The facts module will
+    always collect a base set of facts from the device and can
+    enable or disable collection of additional facts.
+notes:
+  - Tested against EXOS 22.5.1.7
+options:
+  gather_subset:
+    description:
+      - When supplied, this argument will restrict the facts collected
+        to a given subset.  Possible values for this argument include
+        all, hardware, config, and interfaces.  Can specify a list of
+        values to include a larger subset.  Values can also be used
+        with an initial C(M(!)) to specify that a specific subset should
+        not be collected.
+    required: false
+    default: ['!config']
+"""
+
+EXAMPLES = """
+# Collect all facts from the device
```

Comment:
Please use ` - name:` rather than `# ` comment.

Reply:
Fixed, but I probably should have done a normal commit instead of using "git commit --amend" and "git --force push". Apologies...